src/main/homework12/Library.java

```diff
@@ -0,0 +1,126 @@
+                    return;
+                case 3:
+                    listOfBooks(s, library);
+                    break;
+                case 4:
+                    removeBook(s, library);
+                    break;
+                case 5:
+                    listOfAuthors(library);
+                    break;
+                case 6:
+                    bookInfo(s, library);
+                    break;
+
+            }
+
+        }
+
+    }
+
+    public static void addBook(Scanner s, Map<String, List<Book>> library)
+    {
+        System.out.println("Введите идентификатор книги");
+        int newId = s.nextInt();
+        s.nextLine();
+
+        System.out.println("Введите название книги");
+        String newName = s.nextLine();
+
+        System.out.println("Введите описание книги");
+        String newDescription = s.nextLine();
+
+        System.out.println("Введите автора книги");
+        String newNameAuthor = s.nextLine();
+
+        if (library.get(newNameAuthor) == null || library.get(newNameAuthor).isEmpty()) {
+            List<Book> book = new ArrayList<>();
+            book.add(new Book(newId, newName, newDescription, newNameAuthor));
+
+            library.put(newNameAuthor, book);
+
+        } else {
+            List<Book> oldListBook = library.get(newNameAuthor);
+            oldListBook.add(new Book(newId, newName, newDescription, newNameAuthor));
+        }
+    }
+
+    public static void listOfBooks(Scanner s, Map<String, List<Book>> library)
+    {
+        System.out.println("Введите автора");
+        s.nextLine();
+        List<Book> allAuthorBooks = library.get(s.nextLine());
+
+        if (allAuthorBooks == null || allAuthorBooks.isEmpty()) {
+            System.out.println("Для данного автора книг нет");
+        } else System.out.println(allAuthorBooks);
+    }
+
+    public static void removeBook(Scanner s, Map<String, List<Book>> library)
+    {
+        System.out.println("Введите автора");
+        s.nextLine();
+        List<Book> allAuthorBooks1 = library.get(s.nextLine());
+        if (allAuthorBooks1 == null || allAuthorBooks1.isEmpty()) {
+            System.out.println("Для данного автора книг нет");
+        } else System.out.println(allAuthorBooks1);
+        System.out.println("Выберите идентификатор книги для удаления");
+        int idRemove = s.nextInt();
+        for (Book book : allAuthorBooks1) {
+            if (book.getId() == idRemove)
+                allAuthorBooks1.remove(book);
+        }
+        System.out.println(allAuthorBooks1);
+    }
+
+    public static void listOfAuthors(Map<String, List<Book>> library)
+    {
+        Set<String> allAuthors = library.keySet();
+        if (allAuthors == null || allAuthors.isEmpty()) {
+            System.out.println("Авторы отсутствуют");
+        } else
+            System.out.println(allAuthors);
+
+    }
+
+    public static void bookInfo(Scanner s, Map<String, List<Book>> library)
```

Comment:
This method should print book info based on the provided author name and book title